subworkflows/local/align_cellrangermulti.nf

```diff
@@ -0,0 +1,224 @@
+            def data_dict  = meta_clone.find{ it.key == "${meta_clone.feature_type}" }
+            fastqs = data_dict?.value
+            meta_clone.remove( data_dict?.key )
+            [ meta_clone, fastqs ]
+        }
+        .branch {
+            meta, fastq ->
+                gex: meta.feature_type == "gex"
+                    return [ meta, fastq ]
+                vdj: meta.feature_type == "vdj"
+                    return [ meta, fastq ]
+                ab: meta.feature_type == "ab"
+                    return [ meta, fastq ]
+                beam: meta.feature_type == "beam"
+                    return [ meta, fastq ]
+                crispr: meta.feature_type == "crispr"
+                    return [ meta, fastq ]
+                cmo: meta.feature_type == "cmo"
+                    return [ meta, fastq ]
+        }
+        .set { ch_grouped_fastq }
+
+        // Assign other cellranger reference files
+        ch_gex_frna_probeset      = params.gex_frna_probe_set ? file(params.gex_frna_probe_set) : empty_file
+        ch_gex_target_panel       = params.gex_target_panel   ? file(params.gex_target_panel)   : empty_file
+        ch_gex_cmo_set            = params.gex_cmo_set        ? file(params.gex_cmo_set)        : empty_file
+        ch_gex_barcodes           = params.gex_barcode_sample_assignment ? file(params.gex_barcode_sample_assignment) : empty_file
+        ch_fb_reference           = params.fb_reference       ? file(params.fb_reference)       : empty_file
+        ch_vdj_primer_index       = params.vdj_inner_enrichment_primers ? file(params.vdj_inner_enrichment_primers) : empty_file
+        ch_beam_antigen_panel_csv = empty_file // currently not implemented
+        ch_beam_control_panel_csv = empty_file // currently not implemented
+
+        // parse frna and barcode information
+        if (params.cellranger_multi_barcodes) {
+            Channel.fromPath( params.cellranger_multi_barcodes )
+            .splitCsv( header: true )
+            .map {
+                assert it.containsKey('sample'), "The ${params.cellranger_multi_barcodes} samplesheet for cellranger multi must contain a 'sample' colum. Please, check the docs."
+                assert it.containsKey('multiplexed_sample_id'), "The ${params.cellranger_multi_barcodes} samplesheet for cellranger multi must contain a 'multiplexed_sample_id' colum. Please, check the docs."
+                assert it.containsKey('description'), "The ${params.cellranger_multi_barcodes} samplesheet for cellranger multi must contain a 'description' colum. Please, check the docs."
+                assert it.containsKey('probe_barcode_ids') || it.containsKey('cmo_ids'), "The ${params.cellranger_multi_barcodes} samplesheet for cellranger multi must contain at least one, or both, of 'cmo_ids' or 'probe_barcode_ids' columns. Please, check the docs."
+
+                it
+            }
+            .branch {
+                cmo  : it.containsKey('cmo_ids')
```

Comment:
maybe also check if it's not null? I imagine a samplesheet like this could happen in theory

```csv
sample,multiplexed_sample_id,probe_barcode_ids,cmo_ids
sample1,sample1_1,BC301,
sample1,sample1_2,BC302,
sample2,sample2_1,,CMO302
sample2,sample2_2,,CMO303
```

Reply:
Checkings and parsing have migrated to a module-based python script: https://github.com/nf-core/scrnaseq/blob/247-support-for-10x-ffpe-scrna/bin/check_cellrangermulti.py

Which takes those into account.